legal-api/src/legal_api/services/authz.py

```diff
@@ -95,96 +107,441 @@ def has_roles(jwt: JwtManager, roles: List[str]) -> bool:
 ALLOWABLE_FILINGS: Final = {
     'staff': {
         Business.State.ACTIVE: {
-            'adminFreeze': ['SP', 'GP', 'CP', 'BC', 'BEN', 'CC', 'ULC', 'LLC'],
-            'alteration': ['BC', 'BEN', 'ULC', 'CC'],
-            'annualReport': ['CP', 'BEN', 'BC', 'ULC', 'CC'],
-            'changeOfAddress': ['CP', 'BEN', 'BC', 'ULC', 'CC'],
-            'changeOfDirectors': ['CP', 'BEN', 'BC', 'ULC', 'CC'],
-            'changeOfRegistration': ['SP', 'GP'],
-            'consentContinuationOut': ['BC', 'BEN', 'CC', 'ULC'],
-            'conversion': ['SP', 'GP', 'BEN'],
-            'correction': ['CP', 'BEN', 'SP', 'GP', 'BC', 'ULC', 'CC'],
-            'courtOrder': ['SP', 'GP', 'CP', 'BC', 'BEN', 'CC', 'ULC', 'LLC'],
-            'dissolution': ['CP', 'BC', 'BEN', 'CC', 'ULC', 'LLC', 'SP', 'GP'],
-            'incorporationApplication': ['CP', 'BC', 'BEN', 'ULC', 'CC'],
-            'registrarsNotation': ['SP', 'GP', 'CP', 'BC', 'BEN', 'CC', 'ULC', 'LLC'],
-            'registrarsOrder': ['SP', 'GP', 'CP', 'BC', 'BEN', 'CC', 'ULC', 'LLC'],
-            'registration': ['SP', 'GP'],
-            'specialResolution': ['CP'],
-            'transition': ['BC', 'BEN', 'CC', 'ULC'],
+            'adminFreeze': {
+                'legalTypes': ['SP', 'GP', 'CP', 'BC', 'BEN', 'CC', 'ULC', 'LLC'],
+            },
+            'alteration': {
+                'legalTypes': ['BC', 'BEN', 'ULC', 'CC'],
+                'blockerChecks': {
+                    'business': [BusinessBlocker.DEFAULT]
+                }
+            },
```

Comment:
`business` attribute is used to check for business level blocker checks for a given filing. 

`BusinessBlocker.DEFAULT` currently a few(admin freeze, pending filings, specific(correction, alteration) pending filings check.  If more granular checks need to happen, `BusinessBlocker.DEFAULT` can be refactored.  

Additional business level checks can also be added to the `BusinessBlocker` enum.